include/grpcpp/impl/codegen/sync.h

```diff
@@ -19,20 +19,18 @@
 #ifndef GRPCPP_IMPL_CODEGEN_SYNC_H
```

Comment:
can you say why this file needs to change?
(wary about potential public api diffs)

Reply:
It doesn't need to.

Two changes here:
- don't include pthread.h and mutex when GPR_ABSEIL_SYNC because they won't be used
- don't include core_codegen_interface because it is never used

Which one of these gives you pause?

Reply:
A general wariness of breaking code when people update versions. Minimizing changes here means maximizing the chance of someone doing so cleanly (would love to see a friction log from the upgrade you're doing now so we can analyze just how that's gone over the years).

That said, I'm probably ok here; wdyt @markdroth?

Reply:
Just reverted this file instead.